### PR TITLE
Canada Age Amount Credit 

### LINKED ISSUE #20: Age Amount Tax credit

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #20 

[Reference](https://www.canada.ca/en/revenue-agency/services/tax/individuals/topics/about-your-tax-return/tax-return/completing-a-tax-return/deductions-credits-expenses/line-30100-amount.html)

### CODE REVIEW

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    rate:
+      values:
+        2021-01-01: 0
+  - threshold:
+      values:
+        2021-01-01: 38_893
+    rate:
+      values:
+        2021-01-01: 0.15
+  - threshold:
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml:15]: $90,313 is mechanically derived from $38,893 and the credit amount

$38,893 + $7,713/15% = $90,313

So you can remove the $90,313 from both here and the income cap, and then ensure nonnegativity in the Python

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      values:
+        2021-01-01: 90_313 
+    rate:
+      values:
+        2021-01-01: 0
+metadata:
+  type: marginal_rate
+  threshold_unit: currency-CAD
+  amount_unit: currency-CAD
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml:24]: remove (percent is default for marginal_rate)

File: policyengine_canada/variables/gov/cra/tax/income/credits/age_amount/age_amount_credit.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    value_type = float
+    entity = Person
+    label = "Age amount credit"
+    unit = CAD
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        p = parameters(period).gov.cra.tax.income.credits.age_amount_credit
+        age_eligibilty = age >= p.age_eligibility
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/cra/tax/income/credits/age_amount/age_amount_credit.py:14]:
```suggestion
        eligible = age >= p.age_eligibility
```
Can remove `income_eligibility`

File: policyengine_canada/tests/gov/cra/tax/income/credits/age_amount_credit.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+- name: Eligible person with income below the rduction line
+  period: 2022
+  input:
+    total_individual_pre_tax_income: 38_000
+    age: 66
```
Comment (MaxGhenis) [inline policyengine_canada/tests/gov/cra/tax/income/credits/age_amount_credit.yaml:5]: Make this an edge case (65)

And change failures to just below (64)

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  period: year
+  name: Canada_age_amount_higher_credit_amount
+  label: Canada age amount credit higher amount
+  reference:
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/reduction.yaml:28]: add legislative reference

**Review by MaxGhenis — COMMENTED** [review]
very cool usage of scale parameters for reductions! @nikhilwoodruff fyi

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+description: Canada Age Amount Credit cap.
+values:
+  2021-01-01: 7_713
+metadata:
+  type: single_amount
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml:5]: remove

File: changelog_entry.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Canada age amount credit.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
    - Age amount credit.  
```

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  unit: currency-CAD
+  period: year
+  name: Canada_age_amount_credit_cap
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml:8]:
```suggestion
  name: age_amount_credit_cap
```

File: policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  label: Canada age amount credit credit cap
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/cra/tax/income/credits/age_amount_credit/amount/credit_cap.yaml:9]:
```suggestion
  label: Age amount credit credit cap
```

**Review by MaxGhenis — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]
